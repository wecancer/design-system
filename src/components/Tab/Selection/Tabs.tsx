import React, {useEffect, useRef, useContext} from 'react'
import styled from 'styled-components'

import Context from './TabSection.context'

const Container = styled.header`
  display: flex;
  align-items: flex-start;
  margin-bottom: 1.25rem;
`

type Props = {
  children: React.ReactNode
}

const Tabs = ({children}: Props) => {
  const ref = useRef<HTMLElement>(null)
  const {setTabActiveId} = useContext(Context)

  useEffect(() => {
    const target = ref.current?.querySelectorAll('[data-tabtarget]')

    if (target?.length) {
      setTabActiveId(target[0].getAttribute('data-tabtarget'))
    }
  }, [setTabActiveId])

  return <Container ref={ref}>{children}</Container>
}

export default Tabs
