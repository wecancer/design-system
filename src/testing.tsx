import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
