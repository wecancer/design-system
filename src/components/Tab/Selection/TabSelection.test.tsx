import {render} from '../../../testing'

import Tab from './Tab'
import Tabs from './Tabs'
import TabContext from '.'
import TabContent from './TabContent'

describe('<TabSection />', () => {
  it('should render snapshot correctly', () => {
    const {container} = render(
      <TabContext>
        <Tabs>
          <Tab target="0">First tab</Tab>
          <Tab target="1">Middle tab</Tab>
          <Tab target="2">Last tab</Tab>
        </Tabs>
        <TabContent id="0">This is my first tab</TabContent>
        <TabContent id="1">This is my middle tab</TabContent>
        <TabContent id="2">This is my last tab</TabContent>
      </TabContext>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the 5 items in the tab section', () => {
    const {getAllByRole} = render(
      <TabContext>
        <Tabs>
          <Tab target="0">First tab</Tab>
          <Tab target="1">Middle tab</Tab>
          <Tab target="2">Last tab</Tab>
        </Tabs>
        <TabContent id="0">This is my first tab</TabContent>
        <TabContent id="1">This is my middle tab</TabContent>
        <TabContent id="2">This is my last tab</TabContent>
      </TabContext>,
    )

    expect(getAllByRole('tab')).toHaveLength(3)
  })

  it('should have the label item', () => {
    const {getByText} = render(
      <TabContext>
        <Tabs>
          <Tab target="0">First tab</Tab>
          <Tab target="1">Middle tab</Tab>
          <Tab target="2">Last tab</Tab>
        </Tabs>
        <TabContent id="0">This is my first tab</TabContent>
        <TabContent id="1">This is my middle tab</TabContent>
        <TabContent id="2">This is my last tab</TabContent>
      </TabContext>,
    )

    const frist = getByText('First tab')

    expect(frist).toBeInstanceOf(HTMLElement)
  })

  it('should render the first item content by default', () => {
    const {getByText, queryByText} = render(
      <TabContext>
        <Tabs>
          <Tab target="0">First tab</Tab>
          <Tab target="1">Middle tab</Tab>
          <Tab target="2">Last tab</Tab>
        </Tabs>
        <TabContent id="0">This is my first tab</TabContent>
        <TabContent id="1">This is my middle tab</TabContent>
        <TabContent id="2">This is my last tab</TabContent>
      </TabContext>,
    )

    const fristContent = getByText('This is my first tab')
    const midleContent = queryByText('This is my middle tab')
    const lastContent = queryByText('This is my last tab')

    expect(fristContent).toBeInstanceOf(HTMLElement)
    expect(midleContent).toBeNull()
    expect(lastContent).toBeNull()
  })
})
