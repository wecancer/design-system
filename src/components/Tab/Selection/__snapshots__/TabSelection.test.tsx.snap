// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabSection /> should render snapshot correctly 1`] = `
.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 0;
  outline: none;
  border: none;
  background: transparent;
}

.c3 {
  font-size: 1rem;
  font-family: Poppins,sans-serif;
  color: #4E4B66;
  text-align: center;
  padding: 0.75rem 1.25rem;
  border-radius: 1rem;
  position: relative;
  -webkit-transition: all ease 250ms;
  transition: all ease 250ms;
  cursor: pointer;
}

.c3:hover {
  color: #14142B;
  background-color: #EFF0F6;
}

.c3:active {
  color: #14142B;
}

.c3[aria-disabled='true'] {
  opacity: 0.7;
  cursor: default;
}

.c3[aria-disabled='true']:hover {
  background-color: transparent;
}

.c3[aria-selected='true'] {
  color: #0080EB;
}

.c3[aria-selected='true']::after {
  content: '‚óè';
  display: block;
  margin: 0;
  position: absolute;
  bottom: -5px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-bottom: 1.25rem;
}

.c0 {
  font-family: Poppins,sans-serif;
}

<section
  class="c0"
>
  <header
    class="c1"
  >
    <button
      aria-disabled="false"
      aria-selected="true"
      class="c2 c3"
      data-tabtarget="0"
      role="tab"
      type="button"
    >
      First tab
    </button>
    <button
      aria-disabled="false"
      aria-selected="false"
      class="c2 c3"
      data-tabtarget="1"
      role="tab"
      type="button"
    >
      Middle tab
    </button>
    <button
      aria-disabled="false"
      aria-selected="false"
      class="c2 c3"
      data-tabtarget="2"
      role="tab"
      type="button"
    >
      Last tab
    </button>
  </header>
  This is my first tab
</section>
`;
